pkg/test/echo/client/parsedresponse.go

```diff
@@ -173,6 +176,25 @@ func (r ParsedResponses) CheckHostOrFail(t test.Failer, expected string) ParsedR
 	return r
 }
 
+// CheckMTLSForHTTP asserts that mutual TLS was used.
+// Note: this only is detectable for *successful* HTTP based traffic. Other types will always pass
+func (r ParsedResponses) CheckMTLSForHTTP() error {
+	return r.Check(func(i int, response *ParsedResponse) error {
+		if !strings.HasPrefix(response.RequestURL, "http://") &&
+			!strings.HasPrefix(response.RequestURL, "grpc://") &&
+			!strings.HasPrefix(response.RequestURL, "ws://") {
+			// Non-HTTP traffic. Fail open, we cannot check mTLS.
+			return nil
+		}
+		_, f1 := response.RawResponse["X-Forwarded-Client-Cert"]
+		_, f2 := response.RawResponse["x-forwarded-client-cert"] // grpc has different casing
+		if !f1 && !f2 {
```

Comment:
`if f1 || f2 { return nil}`  is easier to understand? positive predicates are preferred.